src/drc/api/serializers.py

```diff
@@ -271,23 +375,40 @@ def update(self, instance, validated_data):
         create a new EnkelvoudigInformatieObject with the same
         EnkelvoudigInformatieObjectCanonical
         """
-        instance.integriteit = validated_data.pop('integriteit', None)
-        instance.ondertekening = validated_data.pop('ondertekening', None)
-
-        validated_data_field_names = validated_data.keys()
-        for field in instance._meta.get_fields():
-            if field.name not in validated_data_field_names:
-                validated_data[field.name] = getattr(instance, field.name)
+        integriteit = validated_data.pop('integriteit', None)
+        ondertekening = validated_data.pop('ondertekening', None)
 
-        validated_data['pk'] = None
-        validated_data['versie'] += 1
+        eio = super().update(instance, validated_data)
+        eio.integriteit = integriteit
+        eio.ondertekening = ondertekening
+        eio.save()
 
-        # Remove the lock from the data from which a new
-        # EnkelvoudigInformatieObject will be created, because lock is not a
-        # part of that model
-        validated_data.pop('lock')
+        # each update - delete previous part files
+        if eio.canonical.bestandsdelen.count() > 0:
+            for part in eio.canonical.bestandsdelen.all():
+                part.inhoud.delete()
+                part.delete()
+
+        if not eio.inhoud and eio.bestandsomvang and eio.bestandsomvang > 0:
```

Comment:
looks like this could/should be a private method called by both `create` and `update`, to reduce code duplication.